spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Shape.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.utils
+
+import scala.reflect.ClassTag
+
+trait Shape {
+  def toSingle(): List[Int] = throw new RuntimeException("Invalid operation")
+  def toMulti(): List[Shape] = throw new RuntimeException("Invalid operation")
+}
+
+case class SingleShape(val value: List[Int]) extends Shape {
```

Comment:
I think we should make Shape classes as immutable. Don't expose the value element. If user need to get it, make a copy

Reply:
copy sounds good.

Reply:
List already being immutable and the returning value is just a copy of the reference, so currently there's no way to change the value of `Shape`?

Reply:
ok